builder.go

```diff
@@ -72,7 +72,11 @@ func (b *Builder) AddOrUpdateConfig(path string, fragment any) (Diagnostics, err
 	var pinner runtime.Pinner
 	defer pinner.Unpin()
 
-	encoder := newMaxEncoder(&pinner)
+	encoder, err := newDefaultEncoder(newMaxEncoderConfig(&pinner))
```

Comment:
It's weird the `newDefaultEncoder` function accepts a configuration... It is no longer "default" I guess?